src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +316,241 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
-            ImmutableList.Create("--job-name"), default, "The name of the submitted job.");
+        /// <param name="entryPoint">The entry point to make a command for.</param>
+        /// <returns>The command corresponding to the given entry point with the validators for that command.</returns>
+        private CommandWithValidators CreateSubmitEntryPointCommand(IEntryPoint entryPoint)
+        {
+            var validators = Enumerable.Empty<ValidateSymbol<CommandResult>>();
 
-        /// <summary>
-        /// The shots option.
-        /// </summary>
-        internal static readonly OptionInfo<int> ShotsOption = new OptionInfo<int>(
-            ImmutableList.Create("--shots"),
-            500,
-            "The number of times the program is executed on the target machine.",
-            validator: result =>
-                int.TryParse(result.Tokens.SingleOrDefault()?.Value, out var value) && value <= 0
-                    ? "The number of shots must be a positive number."
-                    : default);
+            var command = new Command(entryPoint.Name, entryPoint.Summary)
+            {
+                Handler = CommandHandler.Create((ParseResult parseResult, AzureSettings settings) => this.Submit(parseResult, settings, entryPoint))
+            };
+            foreach (var option in entryPoint.Options)
+            {
+                command.AddOption(option);
+            }
 
-        /// <summary>
-        /// The output option.
-        /// </summary>
-        internal static readonly OptionInfo<OutputFormat> OutputOption = new OptionInfo<OutputFormat>(
-            ImmutableList.Create("--output"),
-            OutputFormat.FriendlyUri,
-            "The information to show in the output after the job is submitted.");
+            validators = validators
+                .Concat(AddOptionIfAvailable(command, SubscriptionOption))
+                .Concat(AddOptionIfAvailable(command, ResourceGroupOption))
+                .Concat(AddOptionIfAvailable(command, WorkspaceOption))
+                .Concat(AddOptionIfAvailable(command, this.TargetOption))
+                .Concat(AddOptionIfAvailable(command, StorageOption))
+                .Concat(AddOptionIfAvailable(command, AadTokenOption))
+                .Concat(AddOptionIfAvailable(command, BaseUriOption))
+                .Concat(AddOptionIfAvailable(command, LocationOption))
+                .Concat(AddOptionIfAvailable(command, JobNameOption))
+                .Concat(AddOptionIfAvailable(command, ShotsOption))
+                .Concat(AddOptionIfAvailable(command, OutputOption))
+                .Concat(AddOptionIfAvailable(command, DryRunOption))
+                .Concat(AddOptionIfAvailable(command, VerboseOption))
+                .Concat(MarkOptionsAsMutuallyExclusive(
+                    command,
+                    new[] { BaseUriOption.Aliases.First(), LocationOption.Aliases.First() }));
+
+            return (command, validators.ToList());
```

Comment:
If someone forgets the `ToList()`, this will still compile, but could have unintended consequences since evaluating `validators` has side effects. Can you change `CommandWithValidators` to contain an `IReadOnlyCollection` instead of an `IEnumerable`?

Reply:
I'm not sure what you mean by 'evaluating `validators`' Since each validator is a Func, putting them in a list shouldn't have any side effects. I'm not sure why the preference for `IReadOnlyCollection` or IEnumerable?

Reply:
Sorry, you're right. I confused myself since `AddOptionIfAvailable` has side effects but the returned `IEnumerable` doesn't. Why does it call `ToList()` instead of returning the `IEnumerable` though?

Reply:
The `ToList()` was to evaluate the enumerable so it is only evaluated once.

Reply:
IEnumerable has been replaced.